File: Swift/프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.swift
import Foundation

func solution(_ elements:[Int]) -> Int {
    let length = elements.count
    var sumSet: Set<Int> = Set<Int>()
    let circularArray = elements + elements
    
    for i in 0..<length {
        var sum = 0
        for j in 0..<length {
            sum += circularArray[i+j]
            sumSet.insert(sum)
        }
    }
    
    return sumSet.count
}

File: Java/프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.java
import java.util.*;

class Solution {
    public int solution(int[] elements) {
        int length = elements.length;
        HashSet set = new HashSet<Integer>();
        int[] newElements = new int[length * 2];
        for (int i = 0; i < newElements.length; i++) {
            newElements[i] = elements[i % length];
        }
        
        for (int i = 0; i<length; i++) {
            int sum = 0;
            for (int j = 0; j<length; j++) {
                sum += newElements[i+j];
                set.add(sum);
            }
        }
        
        return set.size();
    }
}